README.md

```diff
@@ -16,6 +16,37 @@ This unit test framework is not officially supported. It is currently in experim
 
 If there's a lot of interest in this framework, then we will continue to improve it and move it to a production ready state. 
 
+## Unit Testing and Functional Testing
```

Comment:
This is super helpful Yennifer, thanks! I have one suggestion: can we move this to docs/testing-terminology.md and link there from the main readme? Cheers!